### PR TITLE
texlab: new port

### PR DESCRIPTION
**Pull request by sainnhe**
#### Description

Texlab is a tex language server written in rust.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] submission

###### Tested on

macOS 12.4 21F79 arm64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by herbygillot** [comment]
This is an excellent submission, @sainnhe.  Just some minor comments:

File: tex/texlab/Portfile [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    windows_x86_64_msvc             0.36.1  c811ca4a8c853ef420abd8592ba53ddbbac90410fab6903b3e79972a631f7680 \
+    yaml-rust                        0.4.5  56c1936c4cc7a1c9ab21a1ebb602eb942ba868cbd44a99cb7cdc5892335e1c85
+
+checksums-append    ${distname}${extract.suffix} \
+                    rmd160  46069222389afbc19f221cca6cbe673f97130251 \
+                    sha256  f7b5300c6fc93d7c339c755c67ea84f9c5daced2fb2735545d01a67ccf2ff770 \
+                    size    7539983
+
+destroot {
+    xinstall -m 755 ${worksrcpath}/target/[cargo.rust_platform]/release/texlab ${destroot}${prefix}/bin/
```
Comment (herbygillot) [inline tex/texlab/Portfile:215]:
```suggestion
    xinstall -m 0755 ${worksrcpath}/target/[cargo.rust_platform]/release/texlab ${destroot}${prefix}/bin/
```

**Comment by sainnhe** [comment]
Thanks! I've fixed it now.

**Comment by sainnhe** [comment]
Btw, is it necessary to add live check? Will the GitHub port group automatically check the new releases?

**Comment by herbygillot** [comment]
> Btw, is it necessary to add live check? Will the GitHub port group automatically check the new releases?

The `github` portgroup does provide livecheck automatically, but the problem is that it will match on any version including beta, alpha or rc versions.  That's why you see ports specifying a pattern to filter those out.

**Pull request merged**